src/plugins/dev_tools/public/dev_tools_icon.tsx

```diff
@@ -3,25 +3,136 @@
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
 import { i18n } from '@osd/i18n';
 import { CoreStart } from 'opensearch-dashboards/public';
+import { MemoryRouter } from 'react-router-dom';
+import { MainApp } from './application';
+import { DevToolApp } from './dev_tool';
+import { DevToolsSetupDependencies } from './plugin';
+import './dev_tools_icon.scss';
+
+export function DevToolsIcon({
+  core,
+  devTools,
+  deps,
+  title,
+}: {
+  core: CoreStart;
+  devTools: readonly DevToolApp[];
+  deps: DevToolsSetupDependencies;
+  title: string;
+}) {
+  const [modalVisible, setModalVisible] = useState(false);
+  const elementRef = useRef<HTMLDivElement | null>(null);
+  const setMountPoint = useCallback((renderFn) => {
+    renderFn(elementRef.current);
+    return () => {};
+  }, []);
+
+  useEffect(() => {
+    if (modalVisible) {
+      document.body.classList.add('noScrollByDevTools');
+    } else {
+      document.body.classList.remove('noScrollByDevTools');
+    }
+
+    return () => {
+      document.body.classList.remove('noScrollByDevTools');
+    };
+  }, [modalVisible]);
 
-export function DevToolsIcon({ core, appId }: { core: CoreStart; appId: string }) {
   return (
-    <EuiToolTip
-      content={i18n.translate('devTools.icon.nav.title', {
-        defaultMessage: 'Developer tools',
-      })}
-    >
-      <EuiButtonIcon
-        aria-label="go-to-dev-tools"
-        iconType="consoleApp"
-        onClick={() => {
-          core.application.navigateToApp(appId);
-        }}
-      />
-    </EuiToolTip>
+    <>
+      <EuiToolTip
+        content={i18n.translate('devTools.icon.nav.title', {
+          defaultMessage: 'Developer tools',
+        })}
+      >
+        <EuiButtonIcon
+          aria-label="go-to-dev-tools"
+          iconType="consoleApp"
+          data-test-subj="openDevToolsModal"
+          onClick={() => {
+            setModalVisible(true);
+          }}
+        />
+      </EuiToolTip>
+      {modalVisible ? (
+        /**
+         * We can not use OuiModal component here because OuiModal uses OuiOverlayMask as its parent node
+         * but overlay mask has a default padding bottom that prevent the modal from covering the whole page.
+         */
+        <EuiOverlayMask className="devToolsOverlayMask" headerZindexLocation="below">
+          <div style={{ width: '100vw', height: '100vh', maxWidth: '100vw' }}>
+            <EuiButtonIcon
+              iconType="cross"
+              onClick={() => setModalVisible(false)}
+              className="euiModal__closeIcon"
+              color="text"
+              aria-label="close modal"
+            />
+            <EuiPanel
+              className="eui-fullHeight"
+              paddingSize="none"
+              hasBorder={false}
+              hasShadow={false}
+            >
+              <EuiFlexGroup direction="column" className="eui-fullHeight devToolsModalContent">
+                <EuiFlexItem grow={false}>
+                  <EuiSpacer />
```

Comment:
nit: the default size for Spacer is large, should we changed to m?

Reply:
Sure, updated.